src/routes/Root/Root.tsx

```diff
@@ -38,8 +42,46 @@ const Root = () => {
     logoutForm.current?.submit();
   };
 
+  const { pathname } = useLocation();
+  const profileIndexPath = t('slugs.profile.index');
+  const ownPagesPath = `/${i18n.language}/${profileIndexPath}`;
+  const routes = [
+    {
+      name: t('profile.preferences'),
+      path: t('slugs.profile.preferences'),
+    },
+    {
+      name: t('profile.favorites'),
+      path: t('slugs.profile.favorites'),
+    },
+    {
+      name: t('profile.competences'),
+      path: t('slugs.profile.competences'),
+    },
+    {
+      name: t('profile.work-history'),
+      path: t('slugs.profile.work-history'),
+    },
+    {
+      name: t('profile.education-history'),
+      path: t('slugs.profile.education-history'),
+    },
+    {
+      name: t('profile.free-time-activities'),
+      path: t('slugs.profile.free-time-activities'),
+    },
+    {
+      name: t('profile.something-else'),
+      path: t('slugs.profile.something-else'),
+    },
+  ].map((route) => ({
+    ...route,
+    path: `${profileIndexPath}/${route.path}`,
+    active: !!matchPath(`${ownPagesPath}/${route.path}`, pathname),
```

Comment:
I think this `active` prop could get ommited, since react router already knows if the route is active. In `RoutesNavigationList.tsx:22` replace the `span` with `NavLink` functionality: `<NavLink ... className={ ({isActive}) => isActive ? 'foo' : 'bar' } />`